### PR TITLE
Add set_regularization

### PR DESCRIPTION
**Pull request by Tyler-D**
Add set_regularization function. The config and model itself is updated correctly. 
The train loss includes softmax loss and regularization loss in my train scripts.

### CODE REVIEW

**Comment by qubvel** [comment]
Hi, @Tyler-D 
What about gamma and beta regularizers in batchnorm layer?
Could you explain more detailed how did you check that regularization is applied? I mean is it possible to have some test to check it?

**Comment by Tyler-D** [comment]
OK. There is no regularizer for Batchnorm layer. I can add it for optional argument (Though I haven't use this kind of regularization in my work).

I checked the *total loss* produced by Keras, which includes softmax loss and all the layer's regularization loss. Also I tracked the *softmax loss* only through metrics option.  And before setting regularizers, *total loss* = *softmax loss*; after setting regularizers, *total loss* > *softmax loss*. 

AFAIK, Keras cannot track regularization loss separately (The model adds regularization loss layer by layer during compile). So I'm not sure how to build up a test for this function. Any suggestion ?

**Comment by qubvel** [comment]
I come up with following test:
```python
...
x = np.ones((1,32,32,3))
y = np.ones((1,32,32,1))

model = ... # load any kind of model
new_model = set_regularization(model, l2(0.01))

# define loss function that return 0
def my_loss(gt, pr)
    return pr * 0

model.compile('Adam', loss=my_loss, metrics=['binary_accuracy'])
new_model.compile('Adam', loss=my_loss, metrics=['binary_accuracy'])

loss_1, _ = model.test_on_batch(x, y)
loss_2, _ = new_model.test_on_batch(x, y)

assert loss_1 == 0.
assert loss_2 > 0.

```
I would be happy if you create `test/test_utils.py` and add this test (see other tests as example) + add regularizers for batchnorm layer. And I will merge this. Thanks in advance.

File: tests/test_utils.py [hunk 1]
```diff
@@ -1,22 +1,26 @@
...
+CASE = (
 
-CASE = ((X1, Y1, Unet('resnet18')))
+    (X1, Y1, MODEL),
+)
 
 
 def _test_regularizer(model, reg_model, x, y):
 
     def zero_loss(pr, gt):
```
Comment (qubvel) [inline tests/test_utils.py]: def zero_loss(gt, pr):

File: segmentation_models/utils.py [hunk 2]
```diff
@@ -131,3 +132,51 @@ def to_tuple(x):
...
+                       kernel_regularizer=None,
+                       bias_regularizer=None,
+                       activity_regularizer=None,
+                       beta_regularizer=None,
+                       gamma_regularizer=None
+                       ):
+    """Set regularizers to all layers
+
+    Note:
+       Returned model's config is upated correctly
```
Comment (qubvel) [inline segmentation_models/utils.py]: up**d**ated

File: segmentation_models/utils.py [hunk 3]
```diff
@@ -131,3 +132,51 @@ def to_tuple(x):
...
+
+    Args:
+        model (``keras.models.Model``): instance of keras model
+        kernel_regularizer(``regularizer`): regularizer of kernels
+        bias_regularizer(``regularizer``): regularizer of bias
+        activity_regularizer(``regularizer``): regularizer of activity
+        gamma_regularizer(``regularizer``): regularizer of gamma of BatchNormalization
+        beta_regularizer(``regularizer``): regularizer of bata of BatchNormalization
```
Comment (qubvel) [inline segmentation_models/utils.py]: bata -> beta

**Comment by Tyler-D** [comment]
Hi @qubvel, despite the typo you point out, do you have any idea about the error produced by CI:
```
ValueError: Tensor Tensor("bn_data/beta:0", shape=(3,), dtype=float32_ref) is not an element of this graph.
```
I ran it correctly in my own environments:
Python 3.5
Keras 2.2.4
Tensorflow 1.9.0

**Comment by qubvel** [comment]
Hi @Tyler-D 
`def zero_loss(gt, pr): ` instead of `def zero_loss(pr, gt):`
reverse elements, first one is gt (ground truth)

**Comment by qubvel** [comment]
That does not help :( I will try to test it on my local pc

**Pull request merged**